src/components/forms/NetworkDevicesForm/edit/NetworkDevicePanel.tsx

```diff
@@ -0,0 +1,392 @@
+        const existingDeviceIndex = devices.findIndex(
+          (d) => d.name === originalDeviceName,
+        );
+        let updatedDevices: FormNetworkDevice[];
+
+        if (existingDeviceIndex !== -1) {
+          updatedDevices = [...devices] as FormNetworkDevice[];
+          updatedDevices[existingDeviceIndex] = deviceData;
+        } else {
+          const originalIndex = devices.findIndex(
+            (d) => d.name === originalDeviceName,
+          );
+          if (originalIndex !== -1) {
+            updatedDevices = [...devices] as FormNetworkDevice[];
+            updatedDevices[originalIndex] = deviceData;
+          } else {
+            updatedDevices = [...devices, deviceData] as FormNetworkDevice[];
+          }
+        }
+
+        parentFormik.setFieldValue("devices", updatedDevices);
+      }
+
+      onSave?.();
+      panelParams.clear();
+    },
+  });
+
+  const selectedNetwork = managedNetworks.find(
+    (n) => n.name === formik.values.network,
+  );
+  const networkAcls = getNetworkAclsByNetworkName(formik.values.network);
+  const allSelectedAcls = formik.values.acls
+    ? formik.values.acls.split(",").filter(Boolean)
+    : [];
+  const selectedAcls = Array.from(new Set(networkAcls.concat(allSelectedAcls)));
+
+  const getAclHelperText = (): ReactNode => {
+    if (supportsNicDeviceAcls(selectedNetwork)) {
+      return networkAcls.length > 0
+        ? "Some ACLs are inherited from the network. They cannot be deselected here."
+        : undefined;
+    }
+    return "Network must be of type OVN to customize ACLs.";
+  };
+
+  const handleCancel = () => {
+    formik.resetForm();
+    notify.clear();
+    panelParams.clear();
+  };
+
+  return (
+    <SidePanel>
+      <SidePanel.Header>
+        <SidePanel.HeaderTitle>{getTitle()}</SidePanel.HeaderTitle>
+      </SidePanel.Header>
+      <NotificationRow className="u-no-padding" />
+      <SidePanel.Content className="u-no-padding">
+        <ScrollableContainer
+          dependencies={[notify.notification]}
+          belowIds={["panel-footer"]}
+        >
+          <Form onSubmit={formik.handleSubmit} stacked>
+            <Input
+              label="Device name"
+              required
+              id="name"
+              {...formik.getFieldProps("name")}
+              type="text"
+              placeholder="Enter device name"
+              error={formik.touched.name && formik.errors.name}
+              disabled={isOverride}
+            />
+
+            <NetworkSelector
+              value={formik.values.network}
+              setValue={(value) => {
+                void formik.setFieldValue("network", value);
+
+                if (!supportsNicDeviceAcls(selectedNetwork)) {
+                  void formik.setFieldValue("acls", "");
+                } else {
+                  const newNetworkAcls = getNetworkAclsByNetworkName(value);
+                  void formik.setFieldValue("acls", newNetworkAcls.join(","));
+                }
```

Comment:
```suggestion
                  const newNetworkAcls = getNetworkAclsByNetworkName(value);
                  void formik.setFieldValue("acls", newNetworkAcls.join(","));
```
If we make the functions handle the empty case, then we can set acls to empty string and this generalizes.

Reply:
`getNetworkAclsByNetworkName` already handles the case where the network name is not part of managedNetworks and the case where the managedNetwork doesn't have ACLs. It won't check the type of the network to know if the network supports IP addresses.